app/src/main/java/com/capstone/weatherapp/WeatherRepository.kt

```diff
@@ -26,23 +26,40 @@ class WeatherRepository(private val db: WeatherDatabase) {
 
     suspend fun refreshCityList() {
         withContext(Dispatchers.IO) {
+            _cityListData.postValue(State.Loading)
             try {
-                val cityList = WeatherApiClient.retrofitService.getCities(CITIES, APIKEY, UNITS)
-                db.cityDao().insert(cityList.convertToCityCacheList())
-            } catch (e: Exception) {}
-            _cityListData.postValue(getNetworkCityList())
+                val cityListResponse = WeatherApiClient.retrofitService.getCities(CITIES, APIKEY, UNITS)
+                if (cityListResponse.isSuccessful) {
+                    val response = cityListResponse.body()!!
+                    _cityListData.postValue(State.Success(response.list))
```

Comment:
we don't want to immediately set this to anything. we want to fetch the data from the db. see below comment